File: examples/auth/session-state.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Inspect auth state and gate work on auth being ready.
export async function sessionState(client: JsBaoClient) {
  // #region example
  const signedIn = client.isAuthenticated(); // boolean

  // Wait until a userId is available. Default timeout 5000ms.
  const userId = await client.waitForUserId({ timeoutMs: 5000 });

  // Wait until authenticated AND offline DBs are ready. Returns the mode.
  const ready = await client.waitForAuthReady({ timeoutMs: 6000 });
  // #endregion example
  return { signedIn, userId, ready };
}

File: examples/auth/session-state.swift
import JsBaoClient

// Inspect auth state and gate work on auth being ready.
// On Swift the waiters take a `timeout` (seconds); `waitForAuthReady`
// returns `(userId, mode)`.
func sessionState(client: JsBaoClient) async throws {
  // #region example
  let signedIn = client.isAuthenticated() // Bool

  // Wait until a userId is available.
  let userId = try await client.waitForUserId(timeout: 5)

  // Wait until authenticated AND offline DBs are ready. Returns the mode.
  let ready = try await client.waitForAuthReady(timeout: 6)
  // #endregion example
  _ = (signedIn, userId, ready)
}
